infra/baseServer_test.go

```diff
@@ -160,10 +177,114 @@ func TestAgentAgentMessage(t *testing.T) {
 	}
 }
 
-func TestAgentListeningSpinner(t *testing.T) {
+func TestAgentListeningSpinnerOpen(t *testing.T) {
+	const numAgents = 3
+	m := make([]infra.AgentGeneratorCountPair[ITestBaseAgent], 1)
+	m[0] = infra.MakeAgentGeneratorCountPair(NewTestAgent, numAgents)
+	server := infra.CreateServer[ITestBaseAgent](m, 1, time.Second, 2)
+	server.Initialise()
+	msg := NewTestMessage()
+	agentMap := server.GetAgentMap()
+	arrayOfIDs := make([]uuid.UUID, numAgents)
+	i := 0
+	for id := range agentMap {
+		arrayOfIDs[i] = id
+		i++
+	}
+	server.BeginAgentListeningSession()
+	server.SendMessage(msg, arrayOfIDs)
+	fmt.Println(arrayOfIDs)
+	//time.Sleep(10*time.Second)
+	server.EndAgentListeningSession()
+	for _, ag := range agentMap {
+		testAgent := ag.(*TestAgent)
+		if testAgent.receivedMessage == false {
+			t.Errorf("agent did not receive a message")
+		}
+	}
+}
 
+func TestAgentListeningSpinnerClosed(t *testing.T) {
+	const numAgents = 3
+	m := make([]infra.AgentGeneratorCountPair[ITestBaseAgent], 1)
+	m[0] = infra.MakeAgentGeneratorCountPair(NewTestAgent, numAgents)
+	server := infra.CreateServer[ITestBaseAgent](m, 1, time.Second, 2)
+	server.Initialise()
+	msg := NewTestMessage()
+	agentMap := server.GetAgentMap()
+	arrayOfIDs := make([]uuid.UUID, numAgents)
+	i := 0
+	for id := range agentMap {
+		arrayOfIDs[i] = id
+		i++
+	}
+	server.BeginAgentListeningSession()
+	server.EndAgentListeningSession()
+	server.SendMessage(msg, arrayOfIDs)
+	fmt.Println(arrayOfIDs)
+	//time.Sleep(10*time.Second)
+
+	for _, ag := range agentMap {
+		testAgent := ag.(*TestAgent)
+		if testAgent.receivedMessage != false {
+			t.Errorf("agent did not stop listening")
+		}
+	}
 }
 
+func TestAgentChannelsClosed(t *testing.T) {
+	defer func() {
+		if panicValue := recover(); panicValue == nil {
+			//check if channels are closed by attempting to send a message to them
+			//if a panic occurs due to closed channel then good
+			t.Errorf("did not close channels")
+		}
+
+		// } else {
+		// 	switch panicV := panicValue.(type) {
+		// 	case string:
+		// 		t.Errorf(panicV)
+
+		// 	default:
+		// 		errmesage := reflect.TypeOf(panicValue)
+		// 		t.Errorf(errmesage.String())
+		// 	}
+		// }
+	}()
+	const numAgents = 3
+	m := make([]infra.AgentGeneratorCountPair[ITestBaseAgent], 1)
+	m[0] = infra.MakeAgentGeneratorCountPair(NewTestAgent, numAgents)
+	server := infra.CreateServer[ITestBaseAgent](m, 1, time.Second, 2)
+	server.Initialise()
+	msg := NewTestMessage()
+	agentMap := server.GetAgentMap()
+	arrayOfIDs := make([]uuid.UUID, numAgents)
+	i := 0
+	for id := range agentMap {
+		arrayOfIDs[i] = id
+		i++
+	}
+	server.BeginAgentListeningSession()
+	server.EndAgentListeningSession()
+	server.Cleanup()
```

Comment:
From reading into `recover()`, most sources seem to say that it's a hacky approach. 

I would consider manually iterating over the channels (since we've now exposed the getters) and running something like `data, ok := <- ch`, as this will yield if the channel is open in the `ok` variable